File: android/core/src/main/kotlin/dev/odeon/android/ui/Cores.kt
package dev.odeon.android.ui

import androidx.compose.ui.graphics.Color

/// A paleta, e ela é a mesma da web.
///
/// Os valores saem letra por letra do `web/src/styles.css`. Não é preciosismo:
/// o Odeon é um produto só visto de dois lugares, e um dourado diferente no
/// celular faria parecer outro app do mesmo dono.
///
/// | | |
/// |---|---|
/// | `fundo` | `--bg` |
/// | `fundoElevado` | `--bg-raised` — o que sobe: cartões, folhas, diálogos |
/// | `fundoAfundado` | `--bg-sunken` — o que desce: trilhos, poços, o fundo do fundo |
/// | `linha` | `--line` |
/// | `texto` | `--fg` |
/// | `textoApagado` | `--fg-muted` |
/// | `destaque` | `--accent` — o filamento aceso |
/// | `destaqueQuente` | `--accent-hot` — o topo da luz: toque e foco |
/// | `destaqueApagado` | `--accent-dim` — o filamento apagado: réguas, bordas |
/// | `perigo` | `--danger` |
/// | `certo` | `--ok` |
///
/// ## Por que ela mora no `:core`, que é o módulo que não desenha — 12/08/2026
///
/// Ela veio do `ui/Tema.kt` do `:app` quando a TV nasceu, e a régua do `:core`
/// aguenta a mudança sem esticar: **uma cor não desenha**. `Color` é um valor —
/// um `inline class` sobre um `ULong` —, do mesmo naipe que um `ItemDaBiblioteca`.
/// Quem desenha é o `Modifier.background` que a recebe, e esse ficou lá.
///
/// O que forçou a mudança foi a alternativa. Com a paleta no `:app`, o `:tv`
/// teria de copiá-la — e o comentário logo acima já diz o que isso custa,
/// escrito antes de haver TV pra provar: «um dourado diferente no celular faria
/// parecer outro app do mesmo dono». Numa TV é pior, porque é onde as duas
/// telas ficam **lado a lado** na mesma sala.
///
/// A cópia também não avisa quando envelhece. O dia em que alguém acertar o
/// `--accent` na `styles.css` e nos dois clientes menos num, esse um continua
/// compilando.
///
/// O que o `:tv` **não** herda é a escala tipográfica: `Tipo` e o `Typography`
/// continuam cada um no seu módulo, porque 11sp de rótulo a três metros de
/// distância não é rótulo, é nada. Mesma tinta, outro tamanho de letra.
///
/// ⚠️ **A `Serifada` desceu depois, e não desmente o parágrafo acima** — ela
/// mora em `Serifada.kt`, ao lado deste arquivo, desde a T0 do
/// `docs/REDESENHO-TV.md` (12/08/2026). A linha que separa as duas é a mesma que
/// separa tinta de tamanho: uma **família de fonte** é a mesma a trinta
/// centímetros e a três metros, e um **corpo de 11sp** não é. Desceu a família;
/// os corpos ficaram.
object Cores {
    val fundo = Color(0xFF0A0A0C)
    val fundoElevado = Color(0xFF131318)
    val fundoAfundado = Color(0xFF050507)
    val linha = Color(0xFF23232C)
    val texto = Color(0xFFECEEF4)
    val textoApagado = Color(0xFF8B8D9A)
    val destaque = Color(0xFFE0B062)
    val destaqueQuente = Color(0xFFFFD98A)
    val destaqueApagado = Color(0xFF8A6A3A)
    val perigo = Color(0xFFFF6B6B)
    val certo = Color(0xFF4ADE80)

    /// ## O papel, e as duas tintas que vivem sobre superfície clara
    ///
    /// Este app é escuro do começo ao fim, e estas três cores são a exceção —
    /// elas existem só onde a tela desenha **um objeto de papel**: a foto de cena
    /// pendurada no varal da ficha, e o bilhete do «continuar».
    ///
    /// ⚠️ **Não são «o tema claro».** Não há tema claro aqui, e nem deve haver:
    /// a sala é escura por decisão, e clarear a interface inteira desfaria o
    /// facho, a cortina e a lâmpada do plano de uma vez. O que estas cores dizem
    /// é outra coisa — que naquele retângulo há papel, e papel não é preto.
    ///
    /// O branco é sujo de propósito (`F2ECE0`, não `FFFFFF`): foto revelada e
    /// ingresso de bilheteria amarelam, e um branco puro no meio de uma tela
    /// escura vira um buraco de luz em vez de um objeto.
    val papel = Color(0xFFF2ECE0)
    val tintaDoPapel = Color(0xFF5C5548)
    val tintaDoBilhete = Color(0xFF241A09)

    /// ⚠️ **O vermelho da cortina é matéria, não significado.**
    ///
    /// O app já tem um vermelho — o `perigo` —, e ele é **semântico**: quer
    /// dizer «isto deu errado» ou «isto vence». O pano do cinema é a primeira
    /// cor deste app que não quer dizer nada: ela é a cor de uma coisa, como a
    /// madeira da prateleira da locadora.
    ///
    /// Por isso são dois nomes próprios e não um `perigo.copy(...)`: no dia em
    /// que alguém mexer no vermelho de erro, a cortina não pode mudar de tom
    /// junto. E no dia em que alguém procurar «que vermelho é este», o nome
    /// responde.
    ///
    /// Os dois tons são a prega: o claro é onde a luz da marquise bate, o fundo
    /// é a dobra. Pano de uma cor só lê como retângulo.
    val cortina = Color(0xFF84252A)
    val cortinaFunda = Color(0xFF4A1114)
}

File: ios/Odeon/Interface/Cores.swift
import SwiftUI

/// A paleta da casa.
///
/// ## ⚠️ Ela não é escolha desta plataforma — é a mesma dos outros dois clientes
///
/// Os cinco valores abaixo saem do `styles.css` da web e do `ui/Tema.kt` do
/// Android, campo a campo. O diagnóstico do redesenho do celular
/// (`android/docs/REDESENHO.md` §1) foi justamente que a cor **já estava certa** e
/// o app ainda parecia outro produto — o que faltava era tipografia, ritmo e
/// objeto, não paleta.
///
/// Ou seja: mexer aqui é divergir de dois clientes de uma vez. Se uma cor
/// precisar mudar, muda nos três.
///
/// | | aqui | web |
/// |---|---|---|
/// | fundo | `#0A0A0C` | `--bg` |
/// | elevado | `#131318` | `--bg-raised` |
/// | texto | `#ECEEF4` | `--fg` |
/// | apagado | `#8B8D9A` | `--fg-muted` |
/// | destaque | `#E0B062` | `--accent` |
///
/// ⚠️ **Não há tema claro, e é decisão.** «O produto é uma sala escura; um tema
/// claro seria uma segunda paleta pra manter sem ninguém pedindo»
/// (`REDESENHO.md` §5). Por isso as telas fixam o esquema escuro em vez de
/// seguir o do sistema — um Odeon branco não é o Odeon.
enum Cores {
    static let fundo = Color(hex: 0x0A0A0C)
    static let fundoElevado = Color(hex: 0x131318)
    static let texto = Color(hex: 0xECEEF4)
    static let textoApagado = Color(hex: 0x8B8D9A)
    static let destaque = Color(hex: 0xE0B062)

    // MARK: - A matéria da locadora

    /// ⚠️ Daqui pra baixo as cores **não querem dizer nada** — elas são a cor de
    /// uma coisa. É a distinção que o `Cores.kt` do Android faz e que vale a pena
    /// copiar inteira: o `destaque` é semântico (isto é tocável, isto é seu), e a
    /// madeira é madeira. No dia em que alguém mexer no dourado da interface, a
    /// prateleira não pode mudar de tom junto.

    /// `--accent-hot`: o topo da luz. É o que a arandela joga na parede.
    static let destaqueQuente = Color(hex: 0xFFD98A)
    /// `--accent-dim`: o filamento apagado — réguas, bordas, o latão da cúpula.
    static let destaqueApagado = Color(hex: 0x8A6A3A)

    /// O papel, e as duas tintas que vivem **sobre superfície clara**.
    ///
    /// ⚠️ Elas existem só onde a tela desenha um **objeto de papel** — a etiqueta
    /// pendurada, a plaquinha da estante. Papel não é preto, e escrever com a cor
    /// da interface em cima dele desfaz o objeto.
    static let papel = Color(hex: 0xF2ECE0)
    static let tintaDoPapel = Color(hex: 0x5C5548)
    static let tintaDoBilhete = Color(hex: 0x241A09)

    /// A madeira da prateleira. Dois tons porque tábua de uma cor só lê como
    /// tarja marrom: o claro é onde a luz da loja bate, o fundo é a sombra.
    static let madeira = Color(hex: 0x6B4A2A)
    static let madeiraFunda = Color(hex: 0x3A2716)
}

extension Color {
    /// `0xE0B062` → a cor. Existe pra a tabela acima poder ser lida do mesmo
    /// jeito que se lê o `styles.css` — conferir hex contra hex é o que mantém os
    /// três clientes iguais, e um `Color(red:green:blue:)` em ponto flutuante
    /// tornaria essa conferência impossível de fazer com o olho.
    init(hex: UInt32) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >> 8) & 0xFF) / 255,
            blue: Double(hex & 0xFF) / 255,
            opacity: 1,
        )
    }
}
